app/services/katello/pulp/repository.rb

```diff
@@ -0,0 +1,23 @@
+module Katello
+  module Pulp
+    class Repository < ::Actions::Pulp::Abstract
+      attr_accessor :repo, :input
+
+      def initialize(repo, input)
+        @repo = repo
+        @input = input
+      end
+
+      def sync
```

Comment:
I think this should pass in a hash of 'overrides', this would include a url override and 'validate'

Reply:
validate as in skip_checksum_validation setting?

Reply:
yes, exactly, in fact all those 'sync_options' checks should be able to be passed in as a hash to override the default behavior.